### ISSUE TITLE
Private fields are not added to classes

### ISSUE THREAD

**Issue by bombsimon** [mirumee/ariadne-codegen#144]
Thanks for this awesome project!

According to the [Pydantic docs](https://docs.pydantic.dev/latest/usage/models/#automatically-excluded-attributes), private fields are automatically excluded from the object.

```python
class MyModel(BaseModel):
    public: str
    _private: str = Field(alias="private")


def main():
    d = {"public": "public", "private": "private"}
    m = MyModel.parse_obj(d) # Same with `MyModel(**d)` or `MyModel(public=..., _pvirate=...)`

    print(m1.public)
    print(m1._private)
```

Will print
```sh
public
default=PydanticUndefined alias='private' alias_priority=2 extra={}
```

Renaming the field to `private` (or anything dropping the leading `_`) makes this work as expected.

This is a general issue for any type that starts with an underscore but mostly interesting for me regarding `__typename` since that's a common practice for GraphQL. I saw you mentioned it in [this](https://github.com/mirumee/ariadne/discussions/605#discussioncomment-898336) discussion but also that the codegen has specific handling for this, f.ex. [here](https://github.com/mirumee/ariadne-codegen/blob/1edb477ae544d4fc825b28cc7c911f36eaeab773/ariadne_codegen/client_generators/result_types.py#L301-L303) and [here](https://github.com/mirumee/ariadne-codegen/blob/1edb477ae544d4fc825b28cc7c911f36eaeab773/ariadne_codegen/codegen.py#L284-L289). This means that accessing `__typename__` on any of the generated classes will only give me the `PydanticUndefined` value.

What's the intended way to access `__typename`? Is this a bug? How would I work with other fields starting with an underscore? For the GraphQL use case I don't see a reason to generate excluded fields, should they be renamed by default?

**Comment by mat-sop** [comment]
Hey, thanks for pointing out this problem. I wasn't aware of this. In my opinion we can handle these fields by stripping leading underscores and keeping the original name as an alias. In addition let's change `__typename__` to something else, like `typename__`. I see that currently `convert_to_snake_case` option already handles some of these cases but not all of them.

**Comment by bombsimon** [comment]
I actually noticed that for single prefixed underscore fields, this is resolved by using `convert_to_snake_case` which I think is enabled by default: https://github.com/mirumee/ariadne-codegen/blob/f61aedd6525744922b2e565a41d783fbf21c37b5/ariadne_codegen/utils.py#L32

So in a query like:

```grapql
query TestQuery {
    __typename
    _a
}
```

By just making the fix for `__typename`, `a` would automatically be renamed to just `a`.

**Comment by mat-sop** [comment]
`_a` -> `a` That the case i was referring to, but `_A` will produce `_a` :) We should take care of this even without `convert_to_snake_case` enabled to not generate "broken" models. It's one of core functionalities so it shouldn't be a plugin.

Regarding implementation, it can be done similar to `iskeyword`, but `process_name` handles more than just fields names. It processes also method names and method's arguments names, I don't think we need to strip `_` from those. In my opinion we should just strip leading `_` from result and input fields without adding anything at the end of them, @rafalp what do you think?

**Comment by bombsimon** [comment]
Right, thanks! I added a commit to drop leading underscores in all class fields and some tests.

**Comment by bombsimon** [comment]
Solved by #148
